rwkv.cpp

```diff
@@ -597,6 +597,20 @@ bool rwkv_eval(struct rwkv_context * ctx, int32_t token, float * state_in, float
     return true;
 }
 
+bool rwkv_eval_array(struct rwkv_context * ctx, int32_t * token_arr, int n_tokens, float * state_in, float * state_out, float * logits_out) {
+    bool success = true;
+    for (int i = 0; i < n_tokens; i++) {
+        success = success && rwkv_eval(ctx, token_arr[i], state_in, state_out, logits_out);
+        state_in = state_out; // switch to using state_out from here onwards
+
+        // If success fail, abort now
+        if( success == false ) {
+            return false;
+        }
+    }
+    return success;
+}
```

Comment:
I propose following implementation:

```c
bool rwkv_eval_sequence(struct rwkv_context * ctx, int32_t * token_sequence, uint32_t n_tokens, float * state_in, float * state_out, float * logits_out) {
    RWKV_ASSERT_FALSE(n_tokens < 1, "Token count must be positive, got %d", n_tokens);

    for (uint32_t i = 0; i < n_tokens; i++) {
        if (!rwkv_eval(ctx, token_sequence[i], i == 0 ? state_in : state_out, state_out, logits_out)) {
            return false;
        }
    }

    return true;
}
```

- it looks simpler to me
- I would prefer to use term `sequence` instead of `array`
- token count made unsigned and with specific size (IDK what exactly `int` is in C and try to avoid it lol)
- added token count correctness check, so users don't try to process zero tokens

Function signature in `rwkv.h` should be changed accordingly.